src/docs/asciidoc/source-sink-connectors/jms.adoc

```diff
@@ -0,0 +1,66 @@
+= JMS
+
+JMS (Java Message Service) connector can be used both as a source and a
+sink for infinite stream processing. The connector is not fault-tolerant.
+On job restart it behaves as if you have started a new job. The source
+does not do snapshotting, the sink does not suppress duplicate data.
+
+In order to use JMS connector, user should include a JMS Client in
+classpath. IO failures are generally handled by JMS Client and do not
+cause the connector to fail. Most of the clients offer a configuration
+parameter to enable auto-reconnection, refer to client documentation for
+details.
+
+== Using JMS as a Source
+
+The JMS Source opens a connection to the JMS Server for each member.
```

Comment:
"JMS Source" -> "JMS source". The word "source" is not a proper name here. Also for "JMS Server", "JMS Topic", "JMS Queue" below.